### ISSUE TITLE
CMake fix to build OpenVDB statically on Windows

### ISSUE THREAD

**Issue by e4lam** [AcademySoftwareFoundation/openvdb#620]
In reviewing the [vcpkg cmake patch for openvdb](https://github.com/microsoft/vcpkg/blob/master/ports/openvdb/0003-fix-cmake.patch), I noticed this change that looks like it might be worthwhile upstreaming?

https://github.com/microsoft/vcpkg/blob/89751b450a161c1451937ee5338e599a51cf53a2/ports/openvdb/0003-fix-cmake.patch#L97
```
 if(WIN32)
-  add_definitions(-D_WIN32 -DNOMINMAX -DOPENVDB_DLL)
+  add_definitions(-D_WIN32 -DNOMINMAX)
+  add_definitions(-D__TBB_NO_IMPLICIT_LINKAGE -D__TBBMALLOC_NO_IMPLICIT_LINKAGE)
+  if (OPENVDB_SHARED)
+    add_definitions(-DOPENEXR_DLL -DOPENVDB_DLL)
+  endif()
 endif()
```

This change is for required building OpenVDB statically on Windows as opposed to always shared. 

I encountered this while trying to help out #603.

**Comment by Idclip** [comment]
These should not live in the OpenVDB config and should be coming from TBBs config. Included this change as part of https://github.com/AcademySoftwareFoundation/openvdb/pull/654. Worth noting that as more packages improve their downstream CMake modules we can look to deprecate the ones bundled with OpenVDB (FindTBB, FindOpenEXR, FindIlmBase will most likely be the first ones)

**Comment by Idclip** [comment]
Solved as part of #708 and #706
